notification/src/handlers/amqpHandler.ts

```diff
@@ -0,0 +1,149 @@
+        const rabbit_password = process.env.AMQP_SERVICE_PWD;
+        const rabit_amqp_url = 'amqp://' + rabbit_usr + ':' + rabbit_password + '@' + rabbit_url;
+
+        console.log("URL: " + rabit_amqp_url)
+
+        var established: boolean = false    // amqp service connection result
+        const handler: AmqpHandler = this   // for ability to access methods and members in callback
+        
+        for (let i = 0; i < retries; i++) {
+            await this.backOff(backoff)
+            await connect(rabit_amqp_url, function (error0: any, connection: Connection) {
+                if (error0) {
+                    console.error(`Error connecting to RabbitMQ: ${error0}.Retrying in ${backoff} seconds`);
+                    return
+                }
+                established = true
+
+                // create the channel
+                handler.initChannel(connection)
+            })
+
+            if (established) {
+                console.log("Connected to RabbitMQ.");
+                break
+            }
+        }
+    }
+
+    /**
+     * Waits for a specific time period.
+     *
+     * @param backOff   Period to wait in seconds
+     */
+    private backOff(backOff: number): Promise<void> {
+        return new Promise(resolve => setTimeout(resolve, backOff * 1000));
+    }
+
+    private initChannel(connection: any) {
+        const handler: AmqpHandler = this   // for ability to access methods and members in callback
+
+        connection.createChannel(function (error1: Error, channel: any) {
+            if (error1) {
+                throw error1;
+            }
+
+            channel.assertQueue(handler.notifQueueName, {
+                durable: false,
+            });
+
+            // Consume from Channel
+            console.log(" [*] Waiting for messages in %s. To exit press CTRL+C", handler.notifQueueName);
+            channel.consume(handler.notifQueueName, handler.handleEvent,{noAck: true}
+            );
+        });
+    }
+
+
+    /**
+     * Handles an event message
+     * @param msg
+     */
+    private handleEvent(msg: any): boolean {
+
+        const eventMessage = JSON.parse(msg.content.toString())
+        const transformationEvent = eventMessage as TransformationEvent
+
+        console.log(`Received Event from Channel: Pipeline id: "${transformationEvent.pipelineId}", 
+        Pipeline Name: "${transformationEvent.pipelineName}`)
+        
+        const isValid = this.isValidTransformationEvent(transformationEvent)
+
+        if (!isValid) {
+            console.error('Message received is not an Transformation Event')
+            return false
+        }
+
+        let handler = new StorageHandler()
+
+        let configs = handler.getConfigsForPipeline(transformationEvent.pipelineId)
+        let executor = new VM2SandboxExecutor()
+        let notificationService = new JSNotificationService(executor)
+
+
+        configs.then(config => {
```

Comment:
Use `await` syntax instead. This means the method will change to `handleEvent(msg: *type*): Promise<void>`